mizu/lib/art/src/sched.rs

```diff
@@ -0,0 +1,88 @@
+use core::{
+    future::Future,
+    hint,
+    sync::atomic::{AtomicUsize, Ordering},
+};
+
+use array_macro::array;
+use arsc_rs::Arsc;
+use async_task::{FallibleTask, Runnable};
+use config::MAX_HARTS;
+use crossbeam_queue::SegQueue;
+use heapless::mpmc::MpMcQueue;
+
+use crate::task::SchedInfo;
+
+static BACKUP: SegQueue<RunTask> = SegQueue::new();
+pub static SCHED: [Scheduler; MAX_HARTS] = array![
+    i => Scheduler {
+        cpu: i,
+        queue: MpMcQueue::new(),
+        count: AtomicUsize::new(0),
+    };
+    MAX_HARTS
+];
+
+struct RunTask {
+    inner: Runnable,
+    info: Arsc<SchedInfo>,
+}
+
+pub struct Scheduler {
```

Comment:
Typical scheduler. However, its the future's responsibility to return to the scheduler, so the actual scheduling algorithm will be written in futures.